chat-widget/src/plugins/createChatTranscript.ts

```diff
@@ -0,0 +1,395 @@
+/* eslint-disable no-useless-escape */
+
+class TranscriptHTMLBuilder {
+    private options: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+    constructor(options: any) {  // eslint-disable-line @typescript-eslint/no-explicit-any
+        this.options = options;
+    }
+
+    createTitleElement() {
+        const htmlData = `<title> ${this.options.title || "Customer Transcript"} </title>`;
+        return htmlData;
+    }
+
+    createExternalScriptElements() {
+        const htmlData = `
+            <script src="https://cdn.botframework.com/botframework-webchat/4.15.7/webchat.js"><\/script>
+            <script src="https://cdnjs.cloudflare.com/ajax/libs/rxjs/7.8.0/rxjs.umd.min.js" integrity="sha512-v0/YVjBcbjLN6scjmmJN+h86koeB7JhY4/2YeyA5l+rTdtKLv0VbDBNJ32rxJpsaW1QGMd1Z16lsLOSGI38Rbg==" crossorigin="anonymous" referrerpolicy="no-referrer"><\/script>
+            <script src="https://unpkg.com/react@18.2.0/umd/react.production.min.js"><\/script>
+            <script src="https://unpkg.com/react-dom@18.2.0/umd/react-dom.production.min.js"><\/script>
+        `;
+
+        return htmlData;
+    }
+
+    createHeadElement() {
+        const htmlData = `
+            <head>
+                ${this.createTitleElement()}
+                ${this.createExternalScriptElements()}
+                <script>
+                    function shareObservable(observable) {
+                        let observers = [];
+                        let subscription;
+
+                        return new Observable(observer => {
+                            if (!subscription) {
+                                subscription = observable.subscribe({
+                                    complete() {
+                                        observers.forEach(observer => observer.complete());
+                                    },
+                                    error(err) {
+                                        observers.forEach(observer => observer.error(err));
+                                    },
+                                    next(value) {
+                                        observers.forEach(observer => observer.next(value));
+                                    }
+                                });
+                            }
+
+                            observers.push(observer);
+
+                            return () => {
+                                observers = observers.filter(o => o !== observer);
+
+                                if (!observers.length) {
+                                    subscription.unsubscribe();
+                                    subscription = null;
+                                }
+                            };
+                        });
+                    }
+                <\/script>
+                <script>
+                    const messages = ${JSON.stringify(this.options.messages)};

```

Comment:
null check !

Reply:
Added null check to assign `[]` to `this.options.message` by default if it is falsy

```
if (!this.options.messages) {
  this.options.messages = [];
}
```